pkg/util/uint256.go

```diff
@@ -55,8 +55,12 @@ func Uint256DecodeBytesBE(b []byte) (u Uint256, err error) {
 
 // Uint256DecodeBytesLE attempts to decode the given string (in LE representation) into a Uint256.
 func Uint256DecodeBytesLE(b []byte) (u Uint256, err error) {
-	b = slice.CopyReverse(b)
-	return Uint256DecodeBytesBE(b)
+	if len(b) != Uint256Size {
+		return u, fmt.Errorf("expected []byte of size %d got %d", Uint256Size, len(b))
+	}
+	u = Uint256(b)
```

Comment:
It'd incur more allocations, we need some place to reverse bytes (`b` can't be changed), that's likely `u`, but then we shouldn't try reversing any input (it can be of a wrong length), so there you have a length check above, now when you've done all of that --- why reusing `Uint256DecodeBytesBE`?